perses/annihilation/ncmc_switching.py

```diff
@@ -165,331 +163,86 @@ def _computeEnergyContribution(self, integrator):
 
         Parameters
         ----------
-        itegrator : NCMCAlchemicalIntegrator subclasses
-            NCMC switching integrator to annihilate or introduce particles alchemically.
-        context : openmm.Context
-            Alchemical context
-        system : simtk.unit.System
-            Real fully-interacting system.
-        initial_positions : simtk.unit.Quantity of dimensions [nparticles,3] with units compatible with angstroms
-            The positions of the alchemical system at the start of the NCMC protocol
-        final_positions : simtk.unit.Quantity of dimensions [nparticles,3] with units compatible with angstroms
-            The positions of the alchemical system at the end of the NCMC protocol
-        direction : str, optional, default='insert'
-            Direction of topology proposal to use for identifying alchemical atoms (allowed values: ['insert', 'delete'])
+        hybrid_thermodynamic_state : openmmtools.states.CompoundThermodynamicState
+            The thermodynamic state of the hybrid sampler.
+        initial_sampler_state : openmmtools.states.SamplerState
+            The sampler state of the nonalchemical system at the start of the NCMC protocol with box vectors
+        final_sampler_state : openmmtools.states.SamplerState
+            The sampler state of the nonalchemical system at the end of the NCMC protocol
 
         Returns
         -------
         logP_energy : float
             The NCMC energy contribution to log probability.
         """
-        logP = integrator.getGlobalVariableByName("final_reduced_potential") - integrator.getGlobalVariableByName("initial_reduced_potential")
+        hybrid_thermodynamic_state.set_alchemical_parameters(0.0)
+        initial_reduced_potential = compute_reduced_potential(hybrid_thermodynamic_state, initial_sampler_state)
```

Comment:
I think I'll stick with the convenience method that I already have, since it ensures that the `context` is correctly updated with the `sampler_state`.